### PR TITLE
PBI2: User can login into the app

### PR DESCRIPTION
**Pull request by nofamex**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,5 @@
 /vendor/
-composer.lock
\ No newline at end of file
+composer.lock
+.phpunit.result.cache
+/coverage/
+/reports/
```
Comment (kanziebub) [inline .gitignore:5]: newline pls ^^

File: src/Migrations/2022_03_06_233456_create_courses_table.php [hunk 2]
```diff
@@ -48,4 +48,4 @@ public function down()
     {
         Schema::dropIfExists($this->tableName);
     }
-}
+}
```
Comment (kanziebub) [inline src/Migrations/2022_03_06_233456_create_courses_table.php]: newline

File: src/Migrations/2022_03_07_092139_create_course_user_table.php [hunk 3]
```diff
@@ -51,4 +51,4 @@ public function down()
     {
         Schema::dropIfExists($this->tableName);
     }
-}
+}
```
Comment (kanziebub) [inline src/Migrations/2022_03_07_092139_create_course_user_table.php]: newline

File: src/Seeders/LMSUserSeeder.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+class LMSUserSeeder extends Seeder
+{
+    public function run()
+    {
+        User::factory()
+            ->count(2)
+            ->create();
+    }
+}
```
Comment (kanziebub) [inline src/Seeders/LMSUserSeeder.php]: you guessed it, newline

Comment (AldiNFitrah) [inline src/Seeders/LMSUserSeeder.php]: newline

File: src/Controllers/AuthController.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    protected function createNewToken($token, $user, $remember = false)
+    {
+        $obj = new stdClass();
+        $obj->access_token = $token;
+        $obj->token_type = 'bearer';
+        $obj->user = $user;
+
+        return ApiResponse::success($obj);
+    }
```
Comment (AldiNFitrah) [inline src/Controllers/AuthController.php]: Is this function has its own url? If not, I think we can move it somewhere outside the controller. Not sure what term we can use, maybe `module` or `service`?

File: src/Migrations/2022_03_06_233456_create_courses_table.php (hunk 2 above)
Comment (AldiNFitrah) [inline src/Migrations/2022_03_06_233456_create_courses_table.php]: newline

File: src/Migrations/2022_03_07_092139_create_course_user_table.php (hunk 3 above)
Comment (AldiNFitrah) [inline src/Migrations/2022_03_07_092139_create_course_user_table.php]: newline

File: src/LMSModule.php [hunk 6]
```diff
@@ -5,7 +5,8 @@
 class LMSModule 
 {
     protected $protected_tables = [
-        'lms_users',
+        'course_user',
+        'courses',
```
Comment (AldiNFitrah) [inline src/LMSModule.php:9]: just to make sure, is it just the table name without the badaso prefix?

File: composer.lock [hunk 7]
```diff
@@ -0,0 +1,7621 @@
+{
```
Comment (FXDROS) [inline composer.lock:1]: I don't think we should push any *.lock files 🤔🤔

**Review by fxdros — COMMENTED** [review]
I think we should not commit and push any `*.lock` files 🤔🤔

File: src/Controllers/AuthController.php (hunk 5 above)
Comment (nofamex) [reply src/Controllers/AuthController.php]: sure i'll move it to a Helper module

File: src/LMSModule.php (hunk 6 above)
Comment (nofamex) [reply src/LMSModule.php:9]: i'll double check

File: .gitignore (hunk 1 above)
Comment (nofamex) [reply .gitignore:5]: i think we dont need a new line in .gitignore file

File: src/LMSModule.php (hunk 6 above)
Comment (nofamex) [reply src/LMSModule.php:9]: yep double checked, its true without the prefix

File: src/Routes/api.php [hunk 8]
```diff
@@ -9,8 +9,9 @@
 
 Route::group(['prefix' => $api_route_prefix, 'as' => 'badaso.', 'middleware' => [ApiRequest::class]], function() {
     Route::group(['prefix' => 'module/lms/v1'], function() {
-        Route::group(['prefix' => 'user'], function() {
-            Route::get('/home', HelpersRoute::getController('UserController@home'));
+        Route::group(['prefix' => 'auth', 'as' => 'auth.'], function() {
+            Route::post('/login', HelpersRoute::getController('AuthController@login'))
+            ->name('login');
```
Comment (AldiNFitrah) [inline src/Routes/api.php]: should be indented 4 more spaces

File: src/Helpers/Token.php [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+class TokenHelper {
+    public static function createNewToken($token, $user, $remember = false)
+    {
+        $obj = new stdClass();
+        $obj->access_token = $token;
+        $obj->token_type = 'bearer';
+        $obj->user = $user;
```
Comment (AldiNFitrah) [inline src/Helpers/Token.php]: use camelCase for variable naming. Don't worry about the json response, badaso ApiResponse will handle it for us.
ref: https://xqsit.github.io/laravel-coding-guidelines/docs/naming-conventions/

**Review by AldiNFitrah — APPROVED** [review]
thanks!

**Review by fxdros — APPROVED** [review]
Thanks

**Review by Zsaschz — APPROVED** [review]

**Pull request merged**